bundle/regal/rules/bugs/inconsistent_args.rego

```diff
@@ -0,0 +1,63 @@
+# METADATA
+# description: Inconsistently named function arguments
+package regal.rules.bugs["inconsistent-args"]
+
+import future.keywords.contains
+import future.keywords.if
+import future.keywords.in
+
+import data.regal.ast
+import data.regal.result
+
+report contains violation if {
+	count(ast.functions) > 0
+
+	# Comprehension indexing — as made obvious here it would be great
+	# to have block level scoped ignore directives...
+	function_args_by_name := {name: args_list |
+		some i
+
+		# regal ignore:prefer-some-in-iteration
+		name := ast.ref_to_string(ast.functions[i].head.ref)
+		args_list := [args |
```

Comment:
We'd possibly get a perf boost by making `args_list` a set.

Reply:
If we didn't need to account for primitives, we could then simply just have made a `count(args_list) > 1` check 😄 .

Reply:
We still need to iterate over it, so I don't think a set would perform better? 🤔